libs/database/src/lib/services/user.service.v2.ts

```diff
@@ -209,4 +211,120 @@ export class UserServiceV2 {
 
     return pages as PageResponseDto[];
   }
+
+  async blockUser(user: User, targetCastcleId: string) {
+    const blockUser = await this.repositoryService
+      .findUser({
+        _id: targetCastcleId,
+      })
+      .exec();
+
+    if (!blockUser || !targetCastcleId)
+      throw CastcleException.USER_OR_PAGE_NOT_FOUND;
+
+    const session = await this._relationshipModel.startSession();
+    await session.withTransaction(async () => {
+      await this.repositoryService.updateRelationship(
+        { user: user._id, followedUser: blockUser._id },
+        {
+          $setOnInsert: {
+            user: user._id,
+            followedUser: blockUser._id,
+            visibility: EntityVisibility.Publish,
+            following: false,
+            blocked: false,
+          },
+          $set: { blocking: true },
+        },
+        { upsert: true }
```

Comment:
```suggestion
        { upsert: true, session }
```
ฝากดูเรื่อง mongoose session เพิ่มหน่อยครับ